From eddee9089c851d24e21d55e78ebbe68b61742336 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 23 Sep 2021 08:32:47 +0300
Subject: [PATCH 09/24] fdts/fsl-s32g274a: Append missing properties instead of
 overwriting

Append clocks settings to mc_cgm0 node instead of overwriting the
entire node.

Issue: ALB-7756
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 fdts/fsl-s32g274a.dtsi | 88 ++++++------------------------------------
 1 file changed, 11 insertions(+), 77 deletions(-)

diff --git a/fdts/fsl-s32g274a.dtsi b/fdts/fsl-s32g274a.dtsi
index e1ce5bbdc..96c30558b 100644
--- a/fdts/fsl-s32g274a.dtsi
+++ b/fdts/fsl-s32g274a.dtsi
@@ -14,83 +14,17 @@
 		     "arm,vexpress,v2p-aarch64", "arm,vexpress";
 };
 
-&clks {
-
-	mc_cgm0: mc_cgm0@40030000 {
-		compatible = "fsl,s32gen1-mc_cgm0";
-		reg = <0x0 0x40030000 0x0 0x3000>;
-
-		assigned-clocks =
-			<&clks S32GEN1_CLK_MC_CGM0_MUX0>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX3>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX4>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX5>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX7>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX8>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX9>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX10>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX12>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX14>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX16>,
-			<&clks S32GEN1_CLK_XBAR_2X>,
-			<&clks S32GEN1_CLK_PER>,
-			<&clks S32GEN1_CLK_FTM0_REF>,
-			<&clks S32GEN1_CLK_FTM1_REF>,
-			<&clks S32GEN1_CLK_CAN_PE>,
-			<&clks S32GEN1_CLK_LIN_BAUD>,
-			<&clks S32GEN1_CLK_GMAC0_TS>,
-			<&clks S32GEN1_CLK_SPI>,
-			<&clks S32GEN1_CLK_SDHC>,
-			<&clks S32GEN1_CLK_QSPI_2X>;
-		assigned-clock-parents =
-			<&clks S32GEN1_CLK_ARM_PLL_DFS1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI2>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI3>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI4>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI5>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_DFS1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_DFS3>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI7>;
-		assigned-clock-rates =
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<S32GEN1_XBAR_2X_FREQ>,
-			<80000000>,
-			<40000000>,
-			<40000000>,
-			<80000000>,
-			<125000000>,
-			<200000000>,
-			<100000000>,
-			<400000000>,
-			/* If S32G274A Rev < 2, CLK_QSPI_2X maximum
-			 * value can be 266666666.
-			 */
-			<400000000>;
-
-		clocks = <&clks S32GEN1_CLK_XBAR_2X>,
-			<&clks S32GEN1_CLK_SERDES_REF>,
-			<&clks S32GEN1_CLK_PER>,
-			<&clks S32GEN1_CLK_FTM0_REF>,
-			<&clks S32GEN1_CLK_FTM1_REF>,
-			<&clks S32GEN1_CLK_CAN_PE>,
-			<&clks S32GEN1_CLK_LIN_BAUD>,
-			<&clks S32GEN1_CLK_SPI>,
-			<&clks S32GEN1_CLK_SDHC>,
-			<&clks S32GEN1_CLK_QSPI_2X>;
-	};
+&mc_cgm0 {
+	clocks = <&clks S32GEN1_CLK_XBAR_2X>,
+		<&clks S32GEN1_CLK_SERDES_REF>,
+		<&clks S32GEN1_CLK_PER>,
+		<&clks S32GEN1_CLK_FTM0_REF>,
+		<&clks S32GEN1_CLK_FTM1_REF>,
+		<&clks S32GEN1_CLK_CAN_PE>,
+		<&clks S32GEN1_CLK_LIN_BAUD>,
+		<&clks S32GEN1_CLK_SPI>,
+		<&clks S32GEN1_CLK_SDHC>,
+		<&clks S32GEN1_CLK_QSPI_2X>;
 };
 
 &ocotp {
-- 
2.17.1

